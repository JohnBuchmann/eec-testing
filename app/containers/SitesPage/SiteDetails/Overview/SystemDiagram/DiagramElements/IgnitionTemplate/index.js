/**
 * Ignition template
 */

import React from 'react';
import { Colors } from 'Theme';
import { getSystemDiagramLabelingConfiguration } from 'Utils/systemDiagram';
import AutoSyncDCBus from '../AutoSyncDCBus';
import Labeling from '../Labeling';

/**
 * Displays Ignition layer template
 * @param {*} props
 */
function IgnitionTemplate() {
  return (
    <>
      {/* Micro Grid Consumption Section frame */}
      <rect
        width="506"
        height="436"
        x="147"
        y="60"
        rx="40"
        ry="40"
        fill="none"
        stroke={Colors.white}
        strokeWidth="1"
      />
      {/* Micro Grid Production Section frame */}
      <rect
        width="489"
        height="436"
        x="660"
        y="61"
        rx="40"
        ry="40"
        fill="none"
        stroke={Colors.white}
        strokeWidth="1"
      />
      <AutoSyncDCBus />

      {getSystemDiagramLabelingConfiguration().map((label) => (
        <Labeling key={label.name} {...label} />
      ))}
    </>
  );
}

export default IgnitionTemplate;
