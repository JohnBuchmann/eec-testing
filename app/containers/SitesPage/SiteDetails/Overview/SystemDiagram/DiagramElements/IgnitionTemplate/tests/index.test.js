import React from 'react';
import { mount } from 'enzyme';
import { IntlProvider } from 'react-intl';
import IgnitionTemplate from '..';

describe('IgnitionTemplate', () => {
  it('should display Ignition Template', () => {
    const actualComponent = mount(
      <svg xmlns="http://www.w3.org/2000/svg">
        <IntlProvider locale="en">
          <IgnitionTemplate />
        </IntlProvider>
      </svg>
    );

    expect(actualComponent).toMatchSnapshot();
  });
});
