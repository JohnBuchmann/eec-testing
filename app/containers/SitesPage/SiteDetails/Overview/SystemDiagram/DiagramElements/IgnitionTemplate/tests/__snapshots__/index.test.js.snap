// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IgnitionTemplate should display Ignition Template 1`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
>
  <IntlProvider
    locale="en"
  >
    <IgnitionTemplate>
      <rect
        fill="none"
        height="436"
        rx="40"
        ry="40"
        stroke="#ffffff"
        strokeWidth="1"
        width="506"
        x="147"
        y="60"
      />
      <rect
        fill="none"
        height="436"
        rx="40"
        ry="40"
        stroke="#ffffff"
        strokeWidth="1"
        width="489"
        x="660"
        y="61"
      />
      <InjectIntl(AutoSyncDCBus)>
        <AutoSyncDCBus
          intl={
            Object {
              "defaultFormats": Object {},
              "defaultLocale": "en",
              "formatDate": [Function],
              "formatHTMLMessage": [Function],
              "formatMessage": [Function],
              "formatNumber": [Function],
              "formatPlural": [Function],
              "formatRelative": [Function],
              "formatTime": [Function],
              "formats": Object {},
              "formatters": Object {
                "getDateTimeFormat": [Function],
                "getMessageFormat": [Function],
                "getNumberFormat": [Function],
                "getPluralFormat": [Function],
                "getRelativeFormat": [Function],
              },
              "locale": "en",
              "messages": Object {},
              "now": [Function],
              "onError": [Function],
              "textComponent": "span",
              "timeZone": null,
            }
          }
        >
          <rect
            data-testid="svg-auto-sync"
            fill="#515e51"
            height={22}
            rx={4}
            stroke="#ffffff"
            width={855}
            x={188}
            y={267}
          />
          <text
            className="makeStyles-autoSyncDCBusTitle-1"
            x={596}
            y={283}
          >
            auto-sync dc-bus
          </text>
        </AutoSyncDCBus>
      </InjectIntl(AutoSyncDCBus)>
      <Labeling
        description="To/From"
        label="T/F"
        x={360}
        y={530}
      >
        <desc>
          Label
        </desc>
        <text
          className="makeStyles-label-3"
          x={360}
          y={530}
        >
          <tspan
            className="makeStyles-labelTitle-4"
          >
            T/F
            :
          </tspan>
          <tspan
            dx={5}
          >
            To/From
          </tspan>
        </text>
      </Labeling>
      <Labeling
        description="Actual"
        label="ACT"
        x={750}
        y={530}
      >
        <desc>
          Label
        </desc>
        <text
          className="makeStyles-label-3"
          x={750}
          y={530}
        >
          <tspan
            className="makeStyles-labelTitle-4"
          >
            ACT
            :
          </tspan>
          <tspan
            dx={5}
          >
            Actual
          </tspan>
        </text>
      </Labeling>
      <Labeling
        description="Device Capacity"
        label="CAP"
        x={951}
        y={530}
      >
        <desc>
          Label
        </desc>
        <text
          className="makeStyles-label-3"
          x={951}
          y={530}
        >
          <tspan
            className="makeStyles-labelTitle-4"
          >
            CAP
            :
          </tspan>
          <tspan
            dx={5}
          >
            Device Capacity
          </tspan>
        </text>
      </Labeling>
    </IgnitionTemplate>
  </IntlProvider>
</svg>
`;
